cellmlmanip/model.py

```diff
@@ -569,36 +581,17 @@ def find_symbols_and_derivatives(self, expression):
                 symbols |= self.find_symbols_and_derivatives(expr.args)
         return symbols
 
-    def set_equation(self, lhs, rhs):
+    def remove_equation(self, equation):
         """
-        Adds an equation defining the variable named in ``lhs``, or replaces an existing one.
-
-        As with :meth:`add_equation()` the LHS must be either a variable symbol or a derivative, and all numbers and
-        variable symbols used in ``lhs`` and ``rhs`` must have been obtained from this model, e.g. via
-        :meth:`add_number()`, :meth:`add_variable()`, or :meth:`get_symbol_by_ontology_term()`.
+        Removes an equation from the model.
 
-        :param lhs: An LHS expression (either a symbol or a derivative).
-        :param rhs: The new RHS expression for this variable.
+        :param equation: The equation to remove.
         """
-        # Get variable symbol named in the lhs
-        lhs_symbol = lhs
-        if lhs_symbol.is_Derivative:
-            lhs_symbol = lhs_symbol.free_symbols.pop()
-        assert isinstance(lhs_symbol, VariableDummy)
-
-        # Check if the variable named in the lhs already has an equation
-        i_existing = None
-        for i, eq in enumerate(self.equations):
-            symbol = eq.lhs.free_symbols.pop() if eq.lhs.is_Derivative else eq.lhs
-            if symbol == lhs_symbol:
-                i_existing = i
-                break
-
-        # Add or replace equation
-        if i_existing is None:
-            self.equations.append(sympy.Eq(lhs, rhs))
-        else:
-            self.equations[i_existing] = sympy.Eq(lhs, rhs)
+        try:
+            i = self.equations.index(equation)
+        except ValueError:
```

Comment:
Would there ever be a scenario where you may try to remove eqs multiple times? E.g. something like
eqs = model.get_equations_for(model.derivative_symbols)) ? Where it could be possible for a variable to be used in more than 1? 
I know this particular example is little silly but can we think of a legit use case?

Reply:
I don't think so! The equations all have to have the form `x = ...`. So there can't be two equations for `x`. Variables can be used lots of times in the RHS, but only in one place in the LHS, so there's never a duplicate equation.